tests/test_service_injection.py

```diff
@@ -0,0 +1,196 @@
+
+    assert fake_endpoint() == {"mock": "record"}
+
+
+def test_dependencies_can_be_configured():
+    """
+    Check that we can pass configuration
+    through the dependency injection framework.
+    """
+
+    @inject(alias=interfaces.Store)
+    class TestStore:
+        """
+        A test store that holds on to whatever kwargs
+        get passed to it
+        """
+
+        def __init__(self, **kwargs):
+            self.kwargs = kwargs
+
+        @staticmethod
+        def _needs_factory() -> bool:
+            return False
+
+    # Instanitate injector with implementations specified
+    inj: Injector = Injector(nop_config, {'store': TestStore})
+
+    # Set configuration
+    inj.configure_service(interfaces.Store, "store", {"kwargy": "foo"})
+
+    @inject
+    def get_me_an_injected_store(store: interfaces.Store) -> interfaces.Store:
+        return store
+
+    # Our TestStore implementation has access to said configuration
+    instantiated_store = get_me_an_injected_store()
+    assert instantiated_store.kwargs['kwargy'] == 'foo'
+
+
+def test_not_a_factory():
+    """
+    Confirm that where a factory is not specified, the
+    service in question is only instantiated once even when
+    an @inject decorated function is called multiple times.
+    """
+
+    @inject(alias=interfaces.Store)
+    class NotAFactoryStore:
+        """
+        A test store thats not using a factory pattern
+        """
+
+        @staticmethod
+        def _needs_factory() -> bool:
+            return False
+
+    # Instanitate injector
+    inj: Injector = Injector(nop_config, {'store': NotAFactoryStore})
+    inj.configure_service(interfaces.Store, "store", {})
+
+    @inject
+    def get_me_an_injected_store(store: interfaces.Store) -> interfaces.Store:
+        return store
+
+    # Instantiating twice gets us exaactly the same object back
+    instantiated_store_1 = get_me_an_injected_store()
+    instantiated_store_2 = get_me_an_injected_store()
+    assert instantiated_store_1 == instantiated_store_2
+
+
+def test_is_a_factory():
+    """
+    Confirm that services where a factory is specified, the
+    service in question is a new object each time an @inject
+    decorated function is called.
+    """
+
+    @inject(alias=interfaces.Store)
+    class IsAFactoryStore:
+        """
+        A test store that is using a factory pattern
+        """
+
+        @staticmethod
+        def _needs_factory() -> bool:
+            return True
+
+    # Instanitate injector
+    inj: Injector = Injector(nop_config, {'store': IsAFactoryStore})
+    inj.configure_service(interfaces.Store, "store", {})
+
+    @inject
+    def get_me_an_injected_store(store: interfaces.Store) -> interfaces.Store:
+        return store
+
+    # Instantiating twice gets us different objects back
+    instantiated_store_1 = get_me_an_injected_store()
+    instantiated_store_2 = get_me_an_injected_store()
+    assert instantiated_store_1 != instantiated_store_2
```

Comment:
Isn't the appropriate way to check whether two references are for the same object using the `is` or `is not` operators?